pageserver/src/walrecord.rs

```diff
@@ -44,6 +44,11 @@ pub enum NeonWalRecord {
         moff: MultiXactOffset,
         members: Vec<MultiXactMember>,
     },
+    /// Update the map of AUX files, either writing or dropping an entry
+    AuxFile {
```

Comment:
I wonder if we should have a more general mechanism where we store the kind in a separate field (`DirectoryKind` from #6736 maybe?). At this point we only have to support aux files but we can add others as well.

Alternatively, we'd add each of these as top-level `NeonWalRecord'`s?

As we serialize this to disk we should decide this now.

`DirectoryKind` has 5 entries, with one having 3 sub-entries, so 7 in total. Of which two are hashmaps (RelDirectory and AuxFilesDirectory) and the rest is hash sets. We can just set content to `Bytes::new()` for hashsets.

Reply:
The trouble with trying to make this generic is that the maps have different key/value types -- so just adding a DirectoryKind here won't make the structure suitable for future use with all the maps.

Rather than make this a (Bytes, Bytes) thing and do another layer of deserialization, I'm inclined to say that we should have each separate type as a separate NeonWalRecord